### ISSUE TITLE
autoscaling/v2beta2 error

### ISSUE THREAD

**Issue by khteh** [kubernetes/kubernetes#73738]
**What happened**:
error: unable to recognize "my-hpa.yml": no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta2"

**What you expected to happen**:
No error according to https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#autoscaling-on-multiple-metrics-and-custom-metrics

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.13.2
- Cloud provider or hardware configuration: AWS EKS
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by liggitt** [comment]
does `kubectl api-versions` show autoscaling/v2beta2 enabled?

**Comment by khteh** [comment]
```
apiVersion: autoscaling/v2beta1
kind: HorizontalPodAutoscaler
metadata:
  name: iconverse-converse-hpa
  namespace: default 
spec:
  minReplicas: 2
  maxReplicas: 5
  scaleTargetRef:
    apiVersion: apps/v1 
    kind: StatefulSet
    name: iconverse-converse
  metrics:
  - type: Resource
    resource:
      name: cpu
      target: 
        type: Utilization
        averageUtilization: 50
status:
  observedGeneration: 1
  lastScaleTime: <some-time>
  currentReplicas: 1
  desiredReplicas: 1
  currentMetrics:
  - type: Resource
    resource:
      name: cpu
      current:
        averageUtilization: 0
        averageValue: 0
```
```
error: error validating "iconverse-converse-hpa.yml": error validating data: [ValidationError(HorizontalPodAutoscaler.spec.metrics[0].resource): unknown field "target" in io.k8s.api.autoscaling.v2beta1.ResourceMetricSource, ValidationError(HorizontalPodAutoscaler.status.currentMetrics[0].resource): unknown field "current" in io.k8s.api.autoscaling.v2beta1.ResourceMetricStatus, ValidationError(HorizontalPodAutoscaler.status.currentMetrics[0].resource): missing required field "currentAverageValue" in io.k8s.api.autoscaling.v2beta1.ResourceMetricStatus, ValidationError(HorizontalPodAutoscaler.status): missing required field "conditions" in io.k8s.api.autoscaling.v2beta1.HorizontalPodAutoscalerStatus]; if you choose to ignore these errors, turn validation off with --validate=false
```

**Comment by MaciekPytel** [comment]
There are several differences between v2beta1 and v2beta2, you can't just take beta2 yaml and change apiVersion. In particular in v2beta1 target is specified differently - instead of 'target' object you have a targetAverageUtilization field.

**Comment by davhin** [comment]
Ran into the same issue, also from the walkthtough referended by @zouyee

**Comment by Guillaume-Mayer** [comment]
Hi.
From which version of kubernetes is the v2beta2 version enabled?
It is in the v1.13 api reference documentation.
But seems it's not enabled, or is there a way to enable it?
Thanks

**Comment by jschuiteboer** [comment]
not only the walkthrough, but also the reference mentions v2beta2: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#horizontalpodautoscaler-v1-autoscaling

server version: v1.13.6-gke.5

`kubectl api-versions | grep autoscaling`:
```
autoscaling/v1
autoscaling/v2beta1
internal.autoscaling.k8s.io/v1alpha1
```

**Comment by bfalese-navent** [comment]
How can i enable autoscaling/v2beta2? i'm on kubernetes 1.13.3

**Comment by bfalese-navent** [comment]
@damemi Could you help us down here? 

Which kubernetes version has enabled autoscaling/v2beta2
Sorry to bother.

**Comment by damemi** [comment]
@bfalese-navent autoscaling/v2beta2 was introduced around 1.12 (https://v1-12.docs.kubernetes.io/docs/setup/release/notes/), so I would think it would be enabled then

**Comment by shinebayar-g** [comment]
https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/ when I apply the following example
```
error validating data: ValidationError(HorizontalPodAutoscaler.spec.metrics[0].resource): unknown field "target" in io.k8s.api.autoscaling.v2beta1.ResourceMetricSource; if you choose to ignore these errors, turn validation off with --validate=false
```

```yaml
apiVersion: autoscaling/v2beta1
kind: HorizontalPodAutoscaler
metadata:
  name: php-apache
  namespace: default
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: php-apache
  minReplicas: 1
  maxReplicas: 10
  metrics:
  - type: Resource
    resource:
      name: cpu
      target:
        type: AverageUtilization
        averageUtilization: 50
```

It literally said `apiVersion: autoscaling/v2beta1` there. I guess as @MaciekPytel mentioned it's changed in `v2beta2` version. But it's mentioned nowhere in docs. And example doc is incorrect :(

v2beta2 API is not available in `v1.13.7-gke.8` as of today.

Also doc mentions
> Notice that you can specify other resource metrics besides CPU. By default, the only other supported resource metric is memory.

So if I want to use memory based autoscaler will it look like this? But when I apply this whole memory part is ignored. 
```yaml
apiVersion: autoscaling/v2beta1
kind: HorizontalPodAutoscaler
metadata:
  name: php-apache
  namespace: default
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: php-apache
  minReplicas: 1
  maxReplicas: 10
  metrics:
  - type: Resource
    resource:
      name: cpu
      targetAverageUtilization: 50
  - type: Resource
    resource:
       name: memory
       targetAverageUtilization: 50
```

**Comment by luisgc93** [comment]
> https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/ need to be updated.

It seems like this hasn't been updated yet right?
